### PR TITLE
Feat/admi get all users

### PR DESCRIPTION
**Pull request by loonita**
This method retrieves all users from the database and excludes their passwords before returning the data

### CODE REVIEW

File: src/user/user.controller.ts [hunk 1]
```diff
@@ -26,16 +29,39 @@ export class UserController {
...
+    throw new BadRequestException('The ID must be a positive number.');
+  }
+  return this.userService.findOne(numericId);
+}
+  
+  @Get('all')
+  @UseGuards(JwtAuthGuard)
+  findAll(@Req() req: Request & { user: any }) {
+    console.log(req.user);
+    if (!req.user) {
```
Comment (LuchoSerruya) [inline src/user/user.controller.ts]: This shouldn't happen - if the JWT guard validates the action, it means the request is authenticated, so you will always have an user, no need to check for it

File: src/user/user.controller.ts [hunk 2]
```diff
@@ -26,16 +29,39 @@ export class UserController {
...
+      throw new BadRequestException('Invalid user data.');
+    }
+    if (req.user.role !== 'admin') {
```
Comment (LuchoSerruya) [inline src/user/user.controller.ts]: It would be better if you use a guard for checking if the user trying to execute the action is an admin or not

**Review by LuchoSerruya — CHANGES_REQUESTED** [review]
Pending changes

File: src/user/user.controller.ts (hunk 1 above)
Comment (loonita) [reply src/user/user.controller.ts]: fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/22/commits/125b085742a7fb2d6dd51253ab45379a574d1456

File: src/user/user.controller.ts (hunk 2 above)
Comment (loonita) [reply src/user/user.controller.ts]: fixed by https://github.com/alkemyTech/CENCO-MIT-Back-T1/pull/22/commits/125b085742a7fb2d6dd51253ab45379a574d1456

**Review by LuchoSerruya — APPROVED** [review]

**Pull request merged**